src/game/Pet.cpp

```diff
@@ -1032,210 +1032,232 @@ bool Pet::CreateBaseAtCreature(Creature* creature)
-
-                        SetBonusDamage(int32(val * 0.15f));
-                        // bonusAP += val * 0.57;
-                        break;
-                    }
-                    case CLASS_MAGE:
-                    {
-                        // 40% damage bonus of mage's frost damage
-                        float val = owner->GetUInt32Value(PLAYER_FIELD_MOD_DAMAGE_DONE_POS + SPELL_SCHOOL_FROST) * 0.4f;
-                        if (val < 0)
-                            val = 0;
-                        SetBonusDamage(int32(val));
-                        break;
-                    }
-                    default:
-                        break;
-                }
+                float scale;
+                if (getLevel() >= cFamily->maxScaleLevel)
+                    scale = cFamily->maxScale;
+                else if (getLevel() <= cFamily->minScaleLevel)
+                    scale = cFamily->minScale;
+                else
+                    scale = cFamily->minScale + float(getLevel() - cFamily->minScaleLevel) / cFamily->maxScaleLevel * (cFamily->maxScale - cFamily->minScale);
+
+                SetObjectScale(scale);
+                UpdateModelData();
             }
 
-            SetBaseWeaponDamage(BASE_ATTACK, MINDAMAGE, float(petlevel - (petlevel / 4)));
-            SetBaseWeaponDamage(BASE_ATTACK, MAXDAMAGE, float(petlevel + (petlevel / 4)));
+            SetMeleeDamageSchool(SpellSchools(SPELL_SCHOOL_NORMAL));
 
-            // SetModifierValue(UNIT_MOD_ATTACK_POWER, BASE_VALUE, float(cinfo->attackpower));
+            if (petlevel < 60)
+                SetUInt32Value(UNIT_FIELD_PETNEXTLEVELEXP, sObjectMgr.GetXPForPetLevel(petlevel));
 
-            PetLevelInfo const* pInfo = sObjectMgr.GetPetLevelInfo(creature_ID, petlevel);
-            if (pInfo)                                      // exist in DB
+            else
             {
-                SetCreateHealth(pInfo->health);
-                SetCreateMana(pInfo->mana);
+                SetUInt32Value(UNIT_FIELD_PETEXPERIENCE, 0);
+                SetUInt32Value(UNIT_FIELD_PETNEXTLEVELEXP, 1000);
+            }
 
-                if (pInfo->armor > 0)
-                    SetModifierValue(UNIT_MOD_ARMOR, BASE_VALUE, float(pInfo->armor));
+            // health, armor and damage modifiers
+            float aMod = 1.0;
+            float hMod = 1.0;
+            float dMod = 1.0;
 
-                for (int stat = 0; stat < MAX_STATS; ++stat)
+            switch(cInfo->Family)
+            {
+                case CREATURE_FAMILY_BAT:
+                case CREATURE_FAMILY_OWL:
+                case CREATURE_FAMILY_SPIDER:
+                case CREATURE_FAMILY_WIND_SERPENT:
                 {
-                    SetCreateStat(Stats(stat), float(pInfo->stats[stat]));
+                    dMod = 1.07;
+                    break;
                 }
+                case CREATURE_FAMILY_BEAR:
+                {
+                    hMod = 1.08;
```

Comment:
what the hell are all those hardcoded value? They should not be placed here. I doubt about that kind of adjustement.